File: leet_easy_c++/leet_189/leet_189b.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        int len = nums.size();
        if(k>=len) k = k%len;
        int dif = len - k;
        vector<int> nums2(nums.begin(), nums.begin()+dif);
        for(int j=0,i=dif; i<len; j++,i++) *(nums.begin()+j) = *(nums.begin()+i);
        for(int j=0, i=k; i<len; j++, i++) *(nums.begin()+i) = *(nums2.begin()+j);
    }
};

File: leet_easy_javascript/leet_189/leet_189b.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var rotate = function(nums, k) {
    var stopper = nums.length - (k%nums.length);
    for(var i=0; i<stopper; i++) nums.push(nums.shift());
};

